src/django_components/autodiscovery.py

```diff
@@ -1,9 +1,15 @@
 import importlib
+import os
 from typing import Callable, List, Optional
 
 from django_components.util.loader import get_component_files
 from django_components.util.logger import logger
 
+# In tests, we want to capture which modules have been loaded, so we can
+# clean them up between tests. But there's no need to track this in
+# production.
+LOADED_MODULES: List[str] = []
```

Comment:
Added logic here so that, while a test is running within `djc_test`, we capture which modules were imported with autodiscover.

This way we can remove the loaded modules after the test, so that autodiscovery works consistently across tests.

If we did not do this, then the second test that runs autodiscovery would still find the same files, but their code would not be executed the second time, because they were already loaded.